lib/game_reader.rb

```diff
@@ -1,3 +1,32 @@
 class GameReader
+  attr_reader :scores
+
+  def initialize(scores = nil)
+    @scores = scores
+    @score_dict ={
+    "A"=>1, "B"=>3, "C"=>3, "D"=>2,
+    "E"=>1, "F"=>4, "G"=>2, "H"=>4,
+    "I"=>1, "J"=>8, "K"=>5, "L"=>1,
+    "M"=>3, "N"=>1, "O"=>1, "P"=>3,
+    "Q"=>10, "R"=>1, "S"=>1, "T"=>1,
+    "U"=>1, "V"=>4, "W"=>4, "X"=>8,
+    "Y"=>4, "Z"=>10
+              }
+  end
+
+   def score(letter)
+    letter = letter.upcase
+    if letter.length > 1
+      letter = letter.split(//)
+      word_score = letter.map { |character| @score_dict[character]}
+      @score = word_score.reduce(:+)
```

Comment:
Like how clear the thinking is in these three lines. It's all very compact, so I think it's probably fine to keep in a single method, but one thing you might want to consider is creating methods that support this `#score` method that will then allow you to unit test each step along the way.

I could see a version of this with `#split_words`, `#score_letters`, and `#sum_letter_scores` as instance methods that you could call from this `#score` method, and the thinking would continue to be *extremely* clear.